File: registers/instruction_register/InstructionRegister.py
from m5.params import *
from m5.SimObject import SimObject

class InstructionRegister(SimObject):
    type = 'InstructionRegister'
    cxx_header = "CAProject/registers/instruction_register/instruction_register.hh"
    cxx_class = "gem5::InstructionRegister"

    inst = Param.UInt16(0, "Instruction stored in the register")

File: registers/instruction_register/instruction_register.cc
#include "CAProject/registers/instruction_register/instruction_register.hh"

namespace gem5
{

InstructionRegister::InstructionRegister(const InstructionRegisterParams &params) : 
    SimObject(params),
    inst(params.inst)
{
}

void
InstructionRegister::setValue(uint16_t val)
{
    inst = val;
}

uint16_t
InstructionRegister::getValue()
{
    return inst;
}

} // namespace gem5
